lib/taste_tester/host.rb

```diff
@@ -188,21 +187,30 @@ def in_test?
 
     def keeptesting
       logger.warn("Renewing taste-tester on #{@name} until" +
-        " #{TasteTester::Config.testing_end_time}")
+        " #{TasteTester::Config.testing_end_time.strftime('%y%m%d%H%M.%S')}")
       if TasteTester::Config.use_ssh_tunnels
         TasteTester::Tunnel.kill(@name)
         @tunnel = TasteTester::Tunnel.new(@name, @server)
         @tunnel.run
       else
         transport = get_transport
-        transport << "touch -t #{TasteTester::Config.testing_end_time}" +
-          " #{TasteTester::Config.timestamp_file}"
+        transport << touchcmd
         transport.run!
       end
     end
 
     private
 
+    def touchcmd
+      touch = Base64.encode64(
+        "if [ 'Darwin' = $(uname) ]; then touch -t \"$(date -r " +
+        "#{TasteTester::Config.testing_end_time.to_i} +'%Y%m%d%H%M.%S')\"" +
+        " #{TasteTester::Config.timestamp_file}; else touch --date \"$(date -d @" +
+        "#{TasteTester::Config.testing_end_time.to_i} +'%Y-%m-%d %T')\"" +
+        " #{TasteTester::Config.timestamp_file}; fi").delete("\n")
```

Comment:
no, this needs to be contiguous string with no newlines, which Base64.encode64 sprinkles all over